api/v1alpha1/policy_helpers.go

```diff
@@ -0,0 +1,36 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	gwapiv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+type PolicyTargetReferences struct {
+	// TargetRef is the name of the Gateway resource this policy
+	// is being attached to.
+	// This Policy and the TargetRef MUST be in the same namespace
+	// for this Policy to have effect and be applied to the Gateway.
+	// TargetRef
+	//
+	// Deprecated: use targetRefs instead
+	TargetRef *gwapiv1a2.LocalPolicyTargetReferenceWithSectionName `json:"targetRef,omitempty"`
+
+	// TargetRefs are the names of the Gateway resources this policy
+	// is being attached to.
+	TargetRefs []gwapiv1a2.LocalPolicyTargetReferenceWithSectionName `json:"targetRefs,omitempty"`
+
+	// TargetSelector contains a selector for resources that this policy
+	// is attached to.
+	TargetSelector map[string]string `json:"targetSelector,omitempty"`
```

Comment:
wouldnt this need more fields so it can also incorporate `Kind` and `Group` ? 
```
      group: gateway.networking.k8s.io
      kind: HTTPRoute
      selector/matchLabels:
        app: foo
```

Reply:
While it's possible to add this, I don't think it's needed.

The selector is checked only against routes and gateways anyways right now, and if a developer/admin wants to limit the matched policies to a specific `kind` it's possible to add the `kind` as a label.

Reply:
as a project, we've tried to use strongly typed fields as much as possible, my suggestion would be to provide a `kind` and `group` field here along with the labelSelector field

Reply:
I think a `kind` and `group` field as part of the target selector defeats the purpose here. If I add a `kind` and `group` field, then how would that work?

Let's say I want to attach to all of my routes (`HTTPRoute`, `GRPCRoute`, `TLSRoute`, and `TCPRoute`) to add some default setting. What do I specify in the `kind` field?

And, considering we enforce that the `group` field is always a single value using CEL, what should I as a user do with the `group` field? Fill it, and make sure it always matches the exactly one valid value? Leave it empty and assume that the default single value is being filled in?

Reply:
reg attaching to all xRoutes, 4 policies could be created per xRoute or 1 policy could be attached to the Gateway

Reply:
sure raised https://github.com/envoyproxy/gateway/issues/3607 to track this work